core/internal/cltest/cltest.go

```diff
@@ -979,7 +979,7 @@ func WaitForPipeline(t testing.TB, nodeID int, jobID int32, expectedPipelineRuns
 
 		var matched []pipeline.Run
 		for _, pr := range prs {
-			if !pr.State.Finished() && pr.State != state {
+			if !pr.State.Finished() || pr.State != state {
```

Comment:
we want to exclude the run if its either incomplete OR if its not at the target state